js/client-topbar.js

```diff
@@ -805,23 +805,40 @@
 	CustomBackgroundPopup.readFile = function (file, popup) {
 		var reader = new FileReader();
 		reader.onload = function (e) {
-			var noSave = false;
 			if (String(e.target.result).length > 4200000) {
 				if (popup) {
 					$('.bgstatus').html('<strong style="background:red;color:white;padding:1px 4px;border-radius:4px;display:block">Image is too large and can\'t be saved. It should be under 3.5MB or so.</strong>');
 				} else {
 					app.addPopupMessage("Image is too large and can't be saved. It should be under 3.5MB or so.");
 				}
-				noSave = true;
 			} else if (popup) {
 				$('.bgstatus').html('Saved');
 				popup.$('.cur').removeClass('cur');
+				Storage.bg.set(e.target.result, 'custom');
+			} else {
+				app.addPopup(ConfirmBackgroundPopup, {bgUrl: e.target.result});
 			}
-			Storage.bg.set(e.target.result, 'custom', noSave);
 		};
 		reader.readAsDataURL(file);
 	};
 
+	var ConfirmBackgroundPopup = this.ConfirmBackgroundPopup = Popup.extend({
+		initialize: function (data) {
+			var buf = '';
+			buf += '<p>Are you sure you want to change your background to this image?</p>';
+			buf += '<p><img src="' + data.bgUrl + '" style="display:block;margin:auto;max-width:90%"></p>';
```

Comment:
Possibly also a max-height if you accidentally drag a super tall but thin image, so the "Cancel" button isn't way off in Narnia.